### PR TITLE
[ENHANCEMENT] Upgrade Angular and Typescript

### LINKED ISSUE #1318: [Enhancement] Upgrade Angular / Typescript to latest versions
**Is your enhancement request related to a problem? Please describe.**
Being out-dated

**Describe the solution you'd like**
Upgrade Angular & Typescript to the latest version

**Describe alternatives you've considered**
RIP

**Additional context**
https://update.angular.io/

### PR DESCRIPTION
**Pull request by jkaltoft**
This PR closes #1318

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Enhancement (to existing content)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] cookbook application / infrastructure changes
- [x] Other... Please describe: Upgrade Angular and Typescript

## Which issue documents the current behaviour?

Issue Number: #1318 

## What is the new behavior?

Angular, Typescript and other dependencies have been upgraded.

## Does this PR introduce a breaking change?

- [x] Yes (probably)
- [ ] No

Don't forget to `npm install`

## Other information

- Update was performed using `--force`. That could leave some incompatible peer dependency issues that may need to be fixed
- Missing peer dependencies
  - Run `npm ls` to see which ones
- Typescript could not be upgraded to latest version due to `@angular/cli`'s TS restrictions.
- Uninstall `node-sass`
  - Run `npm start`
    > 'node-sass' usage is deprecated and will be removed in a future major version. To opt-out of the deprecated behaviour and start using 'sass' uninstall 'node-sass'.
- Decide if action should be taken against warnings when running `npm start`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -123,19 +124,19 @@
...
+    "karma-jasmine-html-reporter": "^1.5.0",
     "karma-junit-reporter": "^2.0.1",
     "karma-spec-reporter": "0.0.32",
     "lint-staged": "^9.4.2",
     "multimatch": "^4.0.0",
-    "ng-mocks": "^9.0.0",
-    "ng-packagr": "^9.0.3",
+    "ng-mocks": "^10.5.4",
+    "ng-packagr": "^11.2.4",
     "node-sass": "^4.13.1",
```
Comment (flemcito) [inline package.json:140]: I think `node-sass` can be bumped to 5.0 - which should allow us to bump our node version > 15

File: package.json [hunk 2]
```diff
@@ -146,10 +147,10 @@
     "stylelint-config-prettier": "^6.0.0",
     "stylelint-prettier": "^1.1.1",
     "ts-node": "5.0.1",
-    "tsickle": "^0.37.0",
-    "tslint": "^6.1.3",
+    "tsickle": "^0.37.1",
+    "tslint": "~6.1.0",
     "tslint-plugin-prettier": "2.0.1",
-    "typescript": "^3.6.5",
+    "typescript": "^4.1.5",
```
Comment (flemcito) [inline package.json:153]: giphub

File: package.json [hunk 3]
```diff
@@ -123,19 +124,19 @@
...
-    "karma-coverage-istanbul-reporter": "^2.1.1",
-    "karma-jasmine": "^3.1.1",
-    "karma-jasmine-html-reporter": "^1.5.2",
+    "jasmine-core": "~3.6.0",
+    "jasmine-spec-reporter": "~5.0.0",
+    "karma": "~6.2.0",
+    "karma-chrome-launcher": "~3.1.0",
+    "karma-coverage-istanbul-reporter": "~3.0.2",
+    "karma-jasmine": "~4.0.0",
+    "karma-jasmine-html-reporter": "^1.5.0",
```
Comment (flemcito) [inline package.json:133]: downgraded? `1.5.2` -> `1.5.0`

File: package.json (hunk 1 above)
Comment (jkaltoft) [reply package.json:140]: It appears that `node-sass` is deprecated. This warning message is displayed when you run `npm start` after the Angular upgrade:

> 'node-sass' usage is deprecated and will be removed in a future major version. To opt-out of the deprecated behaviour and start using 'sass' uninstall 'node-sass'

So, perhaps I should just remove node-sass in this PR

File: package.json (hunk 2 above)
Comment (jkaltoft) [reply package.json:153]: Yeah! Unfortunately I couldn't bump it all the way, but this is great for now

File: package.json (hunk 3 above)
Comment (jkaltoft) [reply package.json:133]: That looks weird. Not sure it matters, but weird.

File: package.json (hunk 1 above)
Comment (flemcito) [reply package.json:140]: If possible, I think that would the best 👍  I found it to be very closely tied to specific node versions, which makes it a nice dependency to kill

File: apps/cookbook/src/app/examples/modal-example/modal-example-configuration/modal-example-configuration.component.ts [hunk 4]
```diff
@@ -40,7 +40,7 @@ export class ModalExampleConfigurationComponent {
   @Input() openFullHeight: boolean;
   @Output() openFullHeightChange = new EventEmitter<boolean>();
 
-  @HostBinding('class.checkbox-xs') true; // Extra small checkboxes
+  @HostBinding('class.checkbox-xs') isCheckboxXs: boolean = true; // Extra small checkboxes
```
Comment (jkaltoft) [inline apps/cookbook/src/app/examples/modal-example/modal-example-configuration/modal-example-configuration.component.ts:43]: @MartinSkovsen

Comment (MartinSkovsen) [reply apps/cookbook/src/app/examples/modal-example/modal-example-configuration/modal-example-configuration.component.ts:43]: See DM from @jakobe. Also I've had the same trouble with that on a different PR in a different component (just setting the hostbinding to true).

File: libs/designsystem/package.json [hunk 5]
```diff
@@ -3,10 +3,11 @@
   "version": "0.0.0",
   "dependencies": {
     "@angular/cdk": "8.2.3",
+    "@fontsource/roboto": "4.2.1",
     "@ionic/angular": "5.5.2",
     "highcharts": "8.0.4",
     "moment": "2.24.0",
-    "@fontsource/roboto": "4.2.1"
+    "tslib": "^2.0.0"
```
Comment (flemcito) [inline libs/designsystem/package.json:10]: I belive this should be a devDependency

Comment (jkaltoft) [reply libs/designsystem/package.json:10]: I think you're right

File: package.json [hunk 6]
```diff
@@ -97,23 +97,24 @@
     "prismjs": "1.23.0",
     "reflect-metadata": "0.1.8",
     "rxjs": "^6.5.4",
-    "zone.js": "~0.10.2"
+    "tslib": "^2.0.0",
```
Comment (jkaltoft) [inline package.json:100]: @flemcito should this be a devDependency?

File: libs/designsystem/package.json (hunk 5 above)
Comment (jkaltoft) [reply libs/designsystem/package.json:10]: https://github.com/kirbydesign/designsystem/pull/1401/files/73c6c835f4114e93ca452739607a7b6c82f27d17#r595833123

File: package.json [hunk 7]
```diff
@@ -77,16 +77,16 @@
   },
   "private": true,
   "dependencies": {
-    "@angular/animations": "^9.1.0",
+    "@angular/animations": "^11.2.5",
     "@angular/cdk": "^8.2.3",
```
Comment (flemcito) [inline package.json:81]: https://www.npmjs.com/package/@angular/cdk
This is angular material.. wondering why we use that. But most current version of this is `11.2.4` - seems like they try to match the angular version

File: package.json [hunk 8]
```diff
@@ -77,16 +77,16 @@
...
-    "@angular/common": "^9.1.0",
-    "@angular/compiler": "^9.1.0",
-    "@angular/core": "^9.1.0",
+    "@angular/common": "^11.2.5",
+    "@angular/compiler": "^11.2.5",
+    "@angular/core": "^11.2.5",
     "@angular/fire": "5.2.1",
```
Comment (flemcito) [inline package.json:85]: @angular/fire is now at `6.1.4`

File: package.json [hunk 9]
```diff
@@ -77,16 +77,16 @@
...
-    "@angular/forms": "^9.1.0",
-    "@angular/platform-browser": "^9.1.0",
-    "@angular/platform-browser-dynamic": "^9.1.0",
-    "@angular/router": "^9.1.0",
+    "@angular/forms": "^11.2.5",
+    "@angular/platform-browser": "^11.2.5",
+    "@angular/platform-browser-dynamic": "^11.2.5",
+    "@angular/router": "^11.2.5",
     "@fontsource/roboto": "4.2.1",
     "@ionic/angular": "^5.5.2",
```
Comment (flemcito) [inline package.json:91]: Small bump, but probably good to do also. @ionic/angular is at `5.6.0`

File: package.json [hunk 10]
```diff
@@ -77,16 +77,16 @@
...
     "@nrwl/angular": "^9.1.2",
```
Comment (flemcito) [inline package.json:92]: @nrwl/angular should probably also be bumped to `11.5.1`

File: package.json (hunk 6 above)
Comment (flemcito) [inline package.json:100]: tslib should be devDependency

File: package.json [hunk 11]
```diff
@@ -97,23 +97,24 @@
...
   "devDependencies": {
     "@angular-builders/custom-webpack": "^8.4.1",
-    "@angular-devkit/build-angular": "~0.901.0",
-    "@angular/cli": "^9.1.0",
-    "@angular/compiler-cli": "^9.1.0",
-    "@ngneat/spectator": "^5.3.1",
+    "@angular-devkit/build-angular": "~0.1102.4",
+    "@angular/cli": "^11.2.4",
+    "@angular/compiler-cli": "^11.2.5",
+    "@ngneat/spectator": "^5.13.4",
```
Comment (flemcito) [inline package.json:108]: @ngneat/spectator is now at `7.1.0`

File: package.json [hunk 12]
```diff
@@ -97,23 +97,24 @@
...
     "@nrwl/cypress": "^9.1.2",
     "@nrwl/workspace": "^9.1.2",
```
Comment (flemcito) [inline package.json:110]: @nrwl/cypress and @nrwl/workspace are both at ´11.5.1´

**Review by flemcito — CHANGES_REQUESTED** [review]
It seems to work, so god work 🥳
But I have added a lot of comments about packages that either depend upon, interact with or in another way relate to angular. I think it is safest to bump these as well as that reduces the risk of incompatibilities (now or in near future)

File: libs/designsystem/src/lib/components/badge/badge.component.ts [hunk 13]
```diff
@@ -6,6 +6,6 @@ import { Component, HostBinding, Input } from '@angular/core';
   styleUrls: ['./badge.component.scss'],
 })
 export class BadgeComponent {
-  @HostBinding('class.kirby-badge') true;
+  @HostBinding('class.kirby-badge') isKirbyBadge: boolean = true;
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/badge/badge.component.ts:9]: Consider moving to static `host` property in `@Component` decorator instead:
```
@Component({
  ...
  // tslint:disable-next-line: no-host-metadata-property
  host: { '[class.kirby-badge]': 'true' },
})
export class BadgeComponent {
...
```

File: libs/designsystem/src/lib/components/chart/chart-helper.ts [hunk 14]
```diff
@@ -1,17 +1,18 @@
...
-import * as exporting from 'highcharts/modules/exporting';
-import * as exportData from 'highcharts/modules/export-data';
-import * as accessibility from 'highcharts/modules/accessibility';
 
 declare var require: any;
 require('highcharts/highcharts-more')(Highcharts);
 require('highcharts/modules/solid-gauge')(Highcharts);
 require('highcharts/modules/broken-axis')(Highcharts);
 
+@Injectable()
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/chart/chart-helper.ts:15]: Why this? 🤔

File: libs/designsystem/src/lib/components/chart/options/activitygauge.ts [hunk 15]
```diff
@@ -82,7 +82,7 @@ export const ActivityGaugeOptions: Options = {
   series: [
     {
       type: 'solidgauge',
-      data: this === undefined ? [] : this.data,
+      data: [],
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/chart/options/activitygauge.ts:85]: This doesn't seem like a safe refactoring... 🔥 
See this PR for an alternative solution: https://github.com/kirbydesign/designsystem/pull/1260

File: libs/designsystem/src/lib/components/chart/options/donut.ts [hunk 16]
```diff
@@ -76,7 +76,7 @@ export const DonutOptions: Options = {
     {
       name: '',
       type: 'pie',
-      data: this === undefined ? [] : this.data,
+      data: [],
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/chart/options/donut.ts:79]: This doesn't seem like a safe refactoring... 🔥 
See this PR for an alternative solution: https://github.com/kirbydesign/designsystem/pull/1260

File: libs/designsystem/src/lib/components/chart/options/areaspline.ts [hunk 17]
```diff
@@ -90,7 +90,7 @@ export const AreaSplineOptions: Options = {
   series: [
     {
       type: 'areaspline',
-      data: this === undefined ? [] : this.data,
+      data: [],
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/chart/options/areaspline.ts:93]: This doesn't seem like a safe refactoring... 🔥 
See this PR for an alternative solution: https://github.com/kirbydesign/designsystem/pull/1260

File: libs/designsystem/src/lib/components/chart/options/timeseries.ts [hunk 18]
```diff
@@ -91,7 +91,7 @@ export const TimeSeriesOptions: Options = {
     {
       type: 'area',
       name: 'USD to EUR',
-      data: this === undefined ? [] : this.data,
+      data: [],
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/chart/options/timeseries.ts:94]: This doesn't seem like a safe refactoring... 🔥 
See this PR for an alternative solution: https://github.com/kirbydesign/designsystem/pull/1260

File: package.json (hunk 6 above)
Comment (flemcito) [reply package.json:100]: I believe so, yes - it's tooling for the typescript compiler, so no need for it run time

File: apps/cookbook/src/app/app-routing.module.ts [hunk 19]
```diff
@@ -7,6 +7,7 @@ import { routes } from './app.routes';
   imports: [
     RouterModule.forRoot(routes, {
       scrollPositionRestoration: 'enabled',
+      relativeLinkResolution: 'legacy',
```
Comment (jakobe) [inline apps/cookbook/src/app/app-routing.module.ts:10]: What's the impact of this? 🤔

File: apps/cookbook/src/app/examples/modal-example/modal-example-configuration/modal-example-configuration.component.ts (hunk 4 above)
Comment (jakobe) [inline apps/cookbook/src/app/examples/modal-example/modal-example-configuration/modal-example-configuration.component.ts:43]: Consider moving to static `host` property in `@Component` decorator instead:
```
@Component({
  ...
  // tslint:disable-next-line: no-host-metadata-property
  host: { '[class.checkbox-xs]': 'true' },
})
export class ModalExampleConfigurationComponent {
...
```

File: libs/designsystem/src/lib/components/icon/icon.component.ts [hunk 20]
```diff
@@ -1,16 +1,16 @@
...
+import { kirbyIconSettings } from './kirby-icon-settings';
 
 @Component({
   selector: 'kirby-icon',
   templateUrl: './icon.component.html',
   styleUrls: ['./icon.component.scss'],
 })
 export class IconComponent implements OnChanges {
-  @HostBinding('class.kirby-icon') true;
+  @HostBinding('class.kirby-icon') isKirbyIcon: boolean = true;
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/icon/icon.component.ts:13]: Consider moving to static `host` property in `@Component` decorator instead:
```
@Component({
  ...
  // tslint:disable-next-line: no-host-metadata-property
  host: { '[class.kirby-icon]': 'true' },
})
export class IconComponent {
...
```

File: libs/designsystem/src/lib/components/list/helpers/list-helper.ts [hunk 21]
```diff
@@ -1,6 +1,9 @@
-import { LoadOnDemandEventData } from '../list.event';
+import { Injectable } from '@angular/core';
+
 import { ListComponent } from '../list.component';
+import { LoadOnDemandEventData } from '../list.event';
 
+@Injectable()
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/list/helpers/list-helper.ts:6]: Is this intentional? 🤔

File: libs/designsystem/src/lib/components/list/list.component.ts [hunk 22]
```diff
@@ -77,7 +77,7 @@ export class ListComponent implements OnInit, AfterViewInit, OnChanges {
    */
   @Input() markSelectedRow = false;
 
-  @HostBinding('class.kirby-list') true;
+  @HostBinding('class.kirby-list') isKirbyList: boolean = true;
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/list/list.component.ts:80]: Consider moving to static `host` property in `@Component` decorator instead:
```
@Component({
  ...
  // tslint:disable-next-line: no-host-metadata-property
  host: { '[class.kirby-list]': 'true' },
})
export class ListComponent {
...
```

File: package.json [hunk 23]
```diff
@@ -97,23 +97,24 @@
...
-    "@types/jasmine": "^3.5.9",
+    "@types/jasmine": "~3.6.0",
     "@types/jasminewd2": "^2.0.8",
     "@types/jest": "^25.1.4",
     "@types/node": "^10.17.17",
```
Comment (flemcito) [inline package.json:114]: @types/node could be bumped to `14.14.35`

File: package.json [hunk 24]
```diff
@@ -146,10 +147,10 @@
     "stylelint-config-prettier": "^6.0.0",
     "stylelint-prettier": "^1.1.1",
     "ts-node": "5.0.1",
```
Comment (flemcito) [inline package.json:149]: ts-node is at `9.1.1`

File: package.json (hunk 7 above)
Comment (jakobe) [reply package.json:81]: @flemcito It's the CDK of Angular Material - not the entire MD.
It was added for - reasons... 🤦🏻 😉 
We'll probably get rid of it when moving to Stencil (and/or re-introduce it with the Virtual Scrolling bug....)
But yes @jkaltoft the versions should be aligned with @angular/core etc.

File: package.json (hunk 9 above)
Comment (jakobe) [reply package.json:91]: @flemcito I would wait with that as to not rock the boat more than necessary - otherwise you won't know whether any regressions are a result of bumping Angular or bumping Ionic... (I agree we should do it soonish, but I don't agree on doing it as part of this PR)

File: package.json (hunk 10 above)
Comment (jakobe) [reply package.json:92]: Agree 👍🏻

File: tsconfig.json [hunk 25]
```diff
@@ -2,7 +2,7 @@
   "compileOnSave": false,
   "compilerOptions": {
     "downlevelIteration": true,
-    "module": "esnext",
+    "module": "es2020",
```
Comment (jakobe) [inline tsconfig.json:5]: What's the impact of this change? Does it only impact internally or also the published package?

**Comment by jkaltoft** [comment]
Thanks a lot, @flemcito. I'll try to bump the dependencies you've mentioned and see if anything breaks

File: package.json [hunk 26]
```diff
@@ -97,23 +97,24 @@
...
+    "@types/jasmine": "~3.6.0",
     "@types/jasminewd2": "^2.0.8",
     "@types/jest": "^25.1.4",
     "@types/node": "^10.17.17",
     "@types/webpack": "^4.41.7",
     "chokidar": "^3.3.1",
-    "codelyzer": "^5.0.1",
+    "codelyzer": "^6.0.0",
     "copy-webpack-plugin": "^5.1.1",
     "core-js": "^2.6.11",
```
Comment (flemcito) [inline package.json:119]: We only import core-js in a single test file - it's a polyfil for modern js features, so I have a hard time believing it is necessary. Kill it with 🔥?

File: package.json [hunk 27]
```diff
@@ -77,16 +77,16 @@
...
   "dependencies": {
-    "@angular/animations": "^9.1.0",
+    "@angular/animations": "^11.2.5",
     "@angular/cdk": "^8.2.3",
-    "@angular/common": "^9.1.0",
-    "@angular/compiler": "^9.1.0",
-    "@angular/core": "^9.1.0",
+    "@angular/common": "^11.2.5",
+    "@angular/compiler": "^11.2.5",
+    "@angular/core": "^11.2.5",
```
Comment (jakobe) [inline package.json:84]: Remember to update the peerDepencies of `libs/designsystem/package.json` as well 🎗️

File: package.json (hunk 26 above)
Comment (jkaltoft) [reply package.json:119]: giphub

Let's see what happens...

File: package.json [hunk 28]
```diff
@@ -123,20 +124,20 @@
     "intersection-observer": "0.7.0",
```
Comment (flemcito) [inline package.json:124]: intersection-observer is at `0.11.0` (not strictly related to angular, but..)

File: tslint.json [hunk 29]
```diff
@@ -7,6 +7,9 @@
   ],
   "rules": {
     "prettier": true,
+    "deprecation": {
+      "severity": "warning"
+    },
```
Comment (jakobe) [inline tslint.json:12]: Which deprecation errors does this change to warnings?

**Review by jakobe — CHANGES_REQUESTED** [review]
Good work! 💪🏻 
- Any reason for doing the update with `--force` flag? 🤔 
- Does changing  `esnext` to `es2020` in `tsconfig.json` have any wider impact, i.e. on projects depending on Kirby? 🤔

File: libs/designsystem/src/lib/components/chart/chart-helper.ts (hunk 14 above)
Comment (jkaltoft) [reply libs/designsystem/src/lib/components/chart/chart-helper.ts:15]: I don't know, but the Angular migration tool added it.

https://update.angular.io/?l=2&v=9.1-10.2

> As of Angular 9, enforcement of @Injectable decorators for DI is stricter and incomplete provider definitions behave differently. *Read more. ng update will migrate you automatically.

*Read more: https://v9.angular.io/guide/migration-injectable

File: libs/designsystem/src/lib/components/chart/options/activitygauge.ts (hunk 15 above)
Comment (jkaltoft) [reply libs/designsystem/src/lib/components/chart/options/activitygauge.ts:85]: Thank you, @jakobe. I was uncomfortable doing that. In the Chart component it looks like it always get overridden, though. But I prefer the solution you mention.

File: apps/cookbook/src/app/app-routing.module.ts (hunk 19 above)
Comment (jkaltoft) [reply apps/cookbook/src/app/app-routing.module.ts:10]: I'm not sure. It was added by the migration tool.

https://angular.io/api/router/ExtraOptions#relativeLinkResolution says

> The default in v11 is `corrected`

so it's a bit odd that it's configured to `legacy`... 🤷‍♂️

File: tslint.json (hunk 29 above)
Comment (jkaltoft) [reply tslint.json:12]: Not sure. It was added by the migration tool. Hard to find any information online.

**Comment by jkaltoft** [comment]
> Good work! 💪🏻
> 
> * Any reason for doing the update with `--force` flag? 🤔

No. No good reasons, other than "it worked" - couldn't make it work without. But perhaps if all dependency issues were addressed before the upgrade it would work.

> * Does changing  `esnext` to `es2020` in `tsconfig.json` have any wider impact, i.e. on projects depending on Kirby? 🤔

Not sure. We can try to change it back to `esnext` and see if it breaks.

File: package.json (hunk 8 above)
Comment (jkaltoft) [reply package.json:85]: Bumping `@angular/fire` to 6.1.4 makes `npm start` fail so I'll postpone that upgrade

File: tslint.json (hunk 29 above)
Comment (jkaltoft) [reply tslint.json:12]: Not sure. We can try to change it back to `esnext` and see if it breaks.

File: libs/designsystem/src/lib/components/chart/chart-helper.ts (hunk 14 above)
Comment (jakobe) [reply libs/designsystem/src/lib/components/chart/chart-helper.ts:15]: Ok, that's sounds fair 👍🏻 
TL;DR: It's basically saying: "You SHOULD ALWAYS decorate services with `@Injectable()` and we plan to make it a requirement in the future, so we'll just help you out and future proof your code base"

File: libs/designsystem/package.json [hunk 30]
```diff
@@ -6,14 +6,13 @@
...
   },
   "peerDependencies": {
-    "@angular/common": ">=8.2.7",
-    "@angular/core": ">=8.2.7",
-    "@angular/forms": ">=8.2.7",
-    "@angular/router": ">=8.2.7",
+    "@angular/common": ">=11.2.5",
+    "@angular/core": ">=11.2.5",
+    "@angular/forms": ">=11.2.5",
+    "@angular/router": ">=11.2.5",
```
Comment (jakobe) [inline libs/designsystem/package.json:15]: On second thought, I'm not sure this is actually required?
I see[ Ionic is only requiring >=8.2.7](https://github.com/ionic-team/ionic-framework/blob/master/angular/package.json#L48-L54)

Comment (jkaltoft) [reply libs/designsystem/package.json:15]: Ok. Should we revert it then?

File: tsconfig.json (hunk 25 above)
Comment (jkaltoft) [reply tsconfig.json:5]: Not sure. We can try to change it back to esnext and see if it breaks.

File: tslint.json (hunk 29 above)
Comment (jkaltoft) [reply tslint.json:12]: Hmm, I found this: https://palantir.github.io/tslint/rules/deprecation/

It's not recently updated (because TS Lint has been deprecated for 2 years 😊) but it _does_ say:

> Any usage of an identifier with the @deprecated JSDoc annotation will trigger a warning. See http://usejsdoc.org/tags-deprecated.html

**Comment by jakobe** [comment]
Replaced by #1441
